applications/GeoMechanicsApplication/custom_processes/apply_k0_procedure_process.cpp

```diff
@@ -78,8 +78,8 @@ void ApplyK0ProcedureProcess::CheckK0MainDirection(const Properties& rProperties
 {
     KRATOS_ERROR_IF(!rProperties.Has(K0_MAIN_DIRECTION))
         << "K0_MAIN_DIRECTION is not defined for element " << ElementId << "." << std::endl;
-    KRATOS_ERROR_IF(rProperties[K0_MAIN_DIRECTION] < 0 || rProperties[K0_MAIN_DIRECTION] > 1)
-        << "K0_MAIN_DIRECTION should be 0 or 1 for element " << ElementId << "." << std::endl;
+    KRATOS_ERROR_IF(rProperties[K0_MAIN_DIRECTION] < 0 || rProperties[K0_MAIN_DIRECTION] > 2)
+        << "K0_MAIN_DIRECTION should be 0, 1 or 2 for element " << ElementId << "." << std::endl;
```

Comment:
For plane strain the applicable directions are 0 and 1, for 3D 0, 1 and 2.
Therefore it would be good if that reflects in the condition for and the text of the error message.

Reply:
Thank you for pointing. Now dimension is used to distinguish an appropriate message.